### PR TITLE
Fix reporting of job dependencies in 'qstat' output  (issue #32)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by boegel** [comment]
@hajgato No test coverage for this?

**Comment by hajgato** [comment]
@boegel I use the original test, which only tries to load the code.

**Comment by hajgato** [comment]
@stdweird I do not understand the test error:
```
files/t/qstat-load.t ....... Bareword "WAIT_DEPENDENCY" not allowed while "strict subs" in use at files/qstat.pl line 263.

Compilation failed in require at files/t/qstat-load.t line 8.
```
 if `WAIT_DEPENDENCY` not allowed, why `JOB_PENDING` is allowed?

**Comment by boegel** [comment]
@hajgato Shouldn't we have a test that verifies that `qstat` produces `H` for a job that depends on another?

**Comment by hajgato** [comment]
@boegel Yes, we should. As well as `qstat` produces `R` for a job that running, as well as ` qstat` produces `C` if the job cancelled, as well as `qstat` produces `Q` for a job that pending, as well as `qstat` produces `S` for a job that was suspended, etc. etc. The current test is the bare minimum, and I feel that testing only for one thing would be meaningless. I do not want to spend days to write an acceptable unit test for ` qstat`, when the whole Perl wrapper will be soon replaced by `jobcli`. ;)

File: files/qstat.pl [hunk 1]
```diff
@@ -255,11 +255,14 @@ sub main
 ################################################################################
 sub stateCode
 {
-    my ($state) = @_;
+    my ($state, $reason) = @_;
 
     if(!defined($state)) {
         return 'U';
     }
+    if((($state & JOB_STATE_BASE) == JOB_PENDING ) && ( $reason == WAIT_DEPENDENCY )) {
```
Comment (stdweird) [inline files/qstat.pl]: add a `&& defined($reason)` before the `&& ($reason == ...)`

**Pull request merged**